src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -203,6 +203,7 @@ protected Project getGitlabProject(GitLabApi gitLabApi) {
                 gitlabProject = gitLabApi.getProjectApi().getProject(projectPath);
                 sshRemote = gitlabProject.getSshUrlToRepo();
                 httpRemote = gitlabProject.getHttpUrlToRepo();
+                projectId = gitlabProject.getId();
```

Comment:
added this for when projects are restored with a project id of still `-1` so any old jobs should also be fixed.